### DISCUSSION TITLE
Manager node in Fault status after upgrade from 2.3.21 to 2.3.52 (bug ?)

### DISCUSSION THREAD

**Discussion by m0nkeyc0de** [Security-Onion-Solutions/securityonion#4493] [Unsupported Versions] [answered]
Hello, my manager node is in Fault status after I upgraded from 2.3.21 to 2.3.52 with soup.

Both Heavy nodes are OK but the Manager node remains in Fault.
One hour after the updates finished, I tried to:
- run "salt-call state.highstate"
- run  "soup" again
- reboot the Manager
But it still remains in the "fault" state, because of "Process Status":

```
Online Since: Jun 11, 2021 3:08 PM
Production EPS: 0
Consumption EPS: 0
Process Status: Fault
Connection Status: OK
Raid Status: Unknown
```

As I'm new to SO I don't really know where to look or if it sounds like a bug. 
Any suggestions ?

**Comment by m0nkeyc0de** [comment]
The 'so-status' command indicates that so-suricata container is in error
```
Checking Docker status

    Docker ------------------------------------------------------- [ OK ]    

Checking container statuses

    ...   
    so-suricata ----------------------------------------------- [ ERROR ]    
    ...    
```
If I try to start it I get a **suricata_state_not_allowed** error ID
```
=========================================================================
Starting suricata...

This could take a while if another Salt job is running. 
Run this command with --force to stop all Salt jobs before proceeding.
=========================================================================
[ERROR   ] Failure!
local:
----------
          ID: suricata_state_not_allowed
    Function: test.fail_without_changes
      Result: False
     Comment: Failure!
     Started: 12:33:41.439925
    Duration: 2.54 ms
     Changes:   

Summary for local
------------
Succeeded: 0
Failed:    1
------------
Total states run:     1
Total run time:   2.540 ms
```

**Comment by m0nkeyc0de** [comment, marked as answer]
I missed the "Support Guidelines" :

- Exact version : 2.3.52
- Install from Security Onion ISO version 2.3.21
- Install type : manager node
- so-status : so-suricata now missing
- salt-call state.highstate --> no failure


After running 'salt-call state.highstate' and 'so-status' again:
Checking container statuses
```
    ...   
    so-suricata --------------------------------------------- [ MISSING ]    
    ...    
```

In https://github.com/Security-Onion-Solutions/securityonion/blob/master/salt/suricata/init.sls we have
```
{% from 'allowed_states.map.jinja' import allowed_states %}
{% if sls in allowed_states %}

... a lot of stuff ...

{% else %}

{{sls}}_state_not_allowed:
  test.fail_without_changes:
    - name: {{sls}}_state_not_allowed

{% endif %}
```
I guest **sls=suricata**

In https://github.com/Security-Onion-Solutions/securityonion/blob/master/salt/allowed_states.map.jinja we have following for **allowed_roles**
```
  {% set allowed_states= salt['grains.filter_by']({
      'so-eval': [
          ...
          'suricata.manager',
          ...
          ],
      'so-heavynode': [
          ...
          'suricata',
          ...
          ],
      'so-helixsensor': [
          ...
          'suricata.manager',
          ...
          ],
      'so-fleet': [
          ...
          ],
      'so-import': [
          ...
          'suricata.manager',
          ...
          'suricata',
          ...
          ],
      'so-manager': [
          ...
          'suricata.manager',
          ...
          ],
      'so-managersearch': [
          ...
          'suricata.manager',
          ],
      'so-node': [
          ...
          ],
      'so-standalone': [
          ...
          'suricata.manager',
          ...
          'suricata',
          ...
          ],
      'so-sensor': [
          ...
          ],
  }, grain='role') %}
```

According to the GUI, the role of my node is "Manager" (I guess the **role** variable value may be **so-manager**).
In the **allowed_states.map.jinja** I only find a **suricata.manager** for the **so-manager** role.
Why does **sls** equal **suricata** for my Manager node ?

On January 20th 2021 a lot of changes were made for #2679 
Could it be an adverse side effect ?

**Reply by m0duspwnens** [reply]
@m0nkeyc0de It appears that the suricata state was run on the manager either directly with`salt-call state.apply suricata` or by using `so-suricata-restart`. We currently have this [issue](https://github.com/Security-Onion-Solutions/securityonion/issues/2977) in to prevent this behavior and it will be resolved in the upcoming release. [https://github.com/Security-Onion-Solutions/securityonion/pull/4543](https://github.com/Security-Onion-Solutions/securityonion/pull/4543)

In the meantime, you can remove so-suricata from being checked by so-status by removing it from `/opt/so/conf/so-status/so-status.conf`.

**Reply by m0nkeyc0de** [reply]
@m0duspwnens Thanks for your answer. I effectively did run  `salt-call state.apply suricata` on the manager node and removing `so-suricata` from  `/opt/so/conf/so-status/so-status.conf` made the alert disappear.
